compiler/ballerina-lang/src/main/resources/compiler.properties

```
@@ -1044,6 +1044,9 @@ error.mismatching.array.literal.values=\
 error.index.out.of.range=\
   index out of range: index: ''{0}''
 
+error.invalid.array.size.reference=\
+  invalid array size reference ''{0}''
```

Let's make this more detailed.

For example, something like
```suggestion
  invalid reference expression ''{0}'' as array size: expected a constant reference expression
```